scripts/irods/test/test_configuration_reload.py

```diff
@@ -0,0 +1,127 @@
+import sys
+
+if sys.version_info < (2,7):
+    import unittest2 as unittest
+else:
+    import unittest
+
+import copy
+import json
+import os
+import signal as signals
+import time
+
+from .. import test
+from . import settings,session
+from .resource_suite import ResourceBase
+from ..controller import IrodsController
+from ..configuration import IrodsConfig
+from .. import lib
+from .. import paths
+from ..test.command import assert_command
+
+try:
+    import zmq
+except ImportError:
+    zmq = None
+
+SessionsMixin = session.make_sessions_mixin([('otherrods','rods')],[])
+
+class TestConfigurationReload(SessionsMixin, unittest.TestCase):
+    def setUp(self):
+        super(TestConfigurationReload, self).setUp()
+        self.admin = self.admin_sessions[0]
+        self.testdir = os.path.join(self.admin.session_collection,'testdir')
+
+    @staticmethod
+    def prepare_rule(property_name):
+        "Prepare a query about the server settings."
+        return f"""
+        msi_get_server_property("{property_name}",*thing);
+        writeLine("stdout","*thing");
+        """
+    @staticmethod
+    def signal_server(server_pid, signal):
+        os.kill(server_pid, signal)
+
+    def test_signal(self):
+        irodsctl = IrodsController()
+        TestConfigurationReload.signal_server(irodsctl.get_server_proc().pid, signals.SIGUSR1)
+        lib.delayAssert(lambda:\
+            lib.log_message_occurrences_greater_than_count(msg="Configuration Reloaded",count=0))
+
+    # Test to see that it recognizes the  configuration's change
+    def test_configuration_change_seen(self):
+        irodsctl = IrodsController()
+        server_config_filename = paths.server_config_path()
+        with lib.file_backed_up(server_config_filename):
+            with open(server_config_filename) as f:
+                svr_cfg = json.load(f)
+            # Here I should make a few different data objects and set their metadata
+            # but for now I'm just curious that it's being seen.
+            svr_cfg["I_am_a_valid_property_name"] = 2
```

Comment:
Consider pulling the property name out into a variable so that it can be used in other places. The same goes for the property value.

Consider using a property value other than `2`. Using `2` could lead to inaccurate test results. Use a string that helps with debugging tests. Many of the iRODS tests will include things in strings like the issue number or a phrase that's related to what is being tested.